pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/SchemaProvider.java

```diff
@@ -31,6 +31,14 @@
      * @param schemaVersion schema version
      * @return schema instance of the provided <tt>schemaVersion</tt>
      */
-    Schema<T> getSchema(byte[] schemaVersion);
+    Schema<T> getSchemaByVersion(byte[] schemaVersion);
```

Comment:
After reading through the code, it seems that we don't actually need a Schema instance here but we only need SchemaInfo. so it might be better to just return `SchemaInfo` here

interface SchemaProvider {

SchemaInfo getSchemaByVersion(byte[] schemaVersion);

SchemaInfo getLatestSchema();

String getTopicName();

}

Reply:
> After reading through the code, it seems that we don't actually need a Schema instance here but we only need SchemaInfo. so it might be better to just return `SchemaInfo` here
> 
> interface SchemaProvider {
> 
> SchemaInfo getSchemaByVersion(byte[] schemaVersion);
> 
> SchemaInfo getLatestSchema();
> 
> String getTopicName();
> 
> }

okey, change it